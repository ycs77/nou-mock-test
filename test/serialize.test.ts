import path from 'node:path'
import { expect, it } from 'vitest'

it('parse pdf data to string', async () => {
  const pdfData = await loadPdf(path.resolve(__dirname, 'fixtures/104-2-linux-1ra-2tx.pdf'))

  const content = pdfDataToString(pdfData)

  expect(content).matchSnapshot()
})

it('calculate max line width', () => {
  const maxWidth = getMaxLineWidth(`
    3.你看過沈國仁的水彩畫作嗎? 水彩是最簡易的繪圖工具，但運用得好，並
    不容易。以你的經驗為例，請說明你駕馭水彩的過程以及從中獲得的成功或
    失敗的經驗。(參閱 1-1)
  `.split('\n').map(line => line.trim()).filter(Boolean).join('\n'))

  expect(maxWidth).toBe(66)
})

it.each([
  '101-1-linux-1ch-2ra-3tx',
  '104-2-linux-1ra-2tx',
  '112-2-linux-1ch-2ra-3tx',
  '112-2-gender-1ch-2ra-3tx',
  '112-2-couplet-1tx',
])('serialize pdf string to paragraphs - %s', async filename => {
  const pdfData = await loadPdf(path.resolve(__dirname, `fixtures/${filename}.pdf`))
  let content = pdfDataToString(pdfData)

  content = serializePdfStringToParagraphs(content)

  expect(content).matchSnapshot()
})
