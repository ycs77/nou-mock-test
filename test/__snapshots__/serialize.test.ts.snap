// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse pdf data to string 1`] = `
"國立空中大學112學年度下學期期中考試題【正參】095
科目：Linux作業系統管理一律橫式作答頁
一、選擇題（每題5分，共50分）
1.下列敘述何者有誤？(A)Ubuntu 作業系統無提供圖形化使用者介面(GUI)的操
作方式(B)Linux作業系統可以安裝在個人電腦或大型電腦上(C)Ubuntu官方
網站提供下載的檔案格式是「ISO」檔(D) Ubuntu Desktop是可以安裝在個人
電腦上的Ubuntu版本。(A，2-1)
2.在 Linux作業系統裡要以文字介面來進行簡單作業，如複製檔案或管理目錄，
可以使用下列哪一個程式? (A) Firefox (B)Impress (C)終端機(D)設定值。(C，
1-6-3)
3.下列何者是在 VirtualBox虛擬機器軟體上安裝Ubuntu 的好處? (A) 不必再另
外準備一台電腦(B)可在一台電腦上安裝多個版本作業系統(C)失敗多次都
可以快速重來 (D)以上皆是。(D，1-4)
4.下列何者是作業系統功能裡記憶體的管理範圍? (A)滑鼠(B)執行中的程式與
資料(C)中央處理器(D)檔案格式。(B，1-2-1)
5.以下關於自由軟體(Freeware)的敘述，何者錯誤? (A)一般公用授權是免費的
(B)允許使用者修改和重新分發軟體的權利(C)使用上無任何限制與規範(D)
其中一種授權狀況是不容許程式以商業化的方式使用或散佈。(C，1-1-1)
6.要 Ubuntu系統裡要調整桌面風格或佈景主題時，可在”設定值”裡選擇哪一個
選項進行? (A)鍵盤(B)電源(C)外觀(D)隱私。(C，2-1-3)
7.下列哪一個是使用者登入 Ubuntu系統後，內定(預設)的工作目錄？(A)根目
前(B)家目錄(C)文件(D)以上皆非。(B，3-2)
8.在 Ubuntu系統裡，下列哪一個功能或指令不是用來管理或安裝套件的? (A)
Ubuntu軟體中心(B)apt (C) synaptic (D) cd。(D，2-3)
9.下列何者不是與檔案與目錄管理相關的指令? (A)ps (B)cd (C)cp (D)ls。(A，3-
3-2)
10.下列哪 Ubuntu目錄是用來存放Linux 核心及開機程式? (A)/bin (B)/root (C)
/boot (D)/etc。(C，1-5)
二、簡答題（每題10分，共50分）
1.請寫出作業系統的四大功能。(1-2-1)
處理器的管理、記憶體的管理、檔案的管理、裝置的管理
2.安裝Ubuntu作業系統有一種方式稱作「Live CD」，請問何謂「Live CD」? (1-4)
3.請簡述Linux作業系統的優點與用途。(1-1-4)
穩定性、輕巧靈活、免費、多人多工的使用環境、硬體規格需求不高、多樣化圖
形介面的加入
4.請列出圖形化使用者介面的五項組成要素。(3-1-1)
桌面、視窗、選單、圖示、按鈕
5.請列出至少四個與目錄或檔案相關的Linux指令，並簡單說明其指令功能。(3-3)
pwd、cd、mkdir、rmdir、ls、cp、rm、mv"
`;

exports[`serialize pdf string to paragraphs - 1011_760008_11 1`] = `
"科目：Linux作業系統管理一律橫式作答2-1頁
一、是非題(20%，每題4分)
(X) 1. Linux作業系統是屬於開放原始碼且為免費的自由軟體，其軟體無任何授權限制。(P.3)
(O) 2. Linux作業系統具有多人多工的優點。(P.9)
(X) 3. Ubuntu作業系統的正體中文官網提供了最新版本作業系統軟體下載，其網址為http://www.ubuntu-tw.com (P.13)
(X) 4. Linux作業系統的目錄結構為樹狀結構，其根目錄為/root。(P.20)
(O) 5. Ubuntu中執行apt-get指令時，必須啟動管理者權限。(P.89)
二、選擇題(20%，每題4分)
(B) 1.下列何者為Linux作業系統正確的USB隨身碟的裝置代號[範圍]? (A)/dev/hd[a-z] (B)/dev/sd[a-z] (C) /dev/fd[a-z] (D) /dev/ud[a-z](P.18)
(D) 2.在Ubuntu中，其輸入法的設定工具名稱為(A) IPod (B) IPus (C) IPad (D) IBus(P.56圖2-11)
(D) 3.開啟Ubuntu中的終端機時，若出現的前導字為「ubuntu@HP-600:~$」，下列敘述何者為「非」? (A) ~代表的是「家目錄」(B) HP-600為主機名稱(C) $為一般使用者提示字元(D) ubuntu為root帳號(P.114)
(D) 4.下列何者指令與顯示檔案內容無關? (A) cat (B) more (C) tac (D) top (P.131)
(C) 5.若某一執行程序之nice值為0，若要將其執行優先順序調高，則應將nice值調為? (A)10 (B) 15 (C) -15 (D) -25(P.167)背面尚有試題
科目：Linux作業系統管理【正參】一律橫式作答2-2頁
三、簡答題(30%，每題10分)
1.請列出三種Linux發佈的版本。
答：參見課本P.6-7
2.請列出Ubuntu中的任三種型態的網路連線設定。(P.64圖2-23)
答：參見課本P.64圖2-23
(有線網路、無線網路、行動寬頻、VPN、DSL )
3.在文件搜尋的指令中有find、whereis與locate，請簡述在搜尋速度上有何差異?。
答：參見課本P.139
find:速度慢，硬碟存取頻繁，比較耗資源
whereis與locate:利用資訊庫搜尋，速度快
四、問答題(30%，每題10分)
1.安裝Ubuntu作業系統可以選擇一種稱作「Live CD」的方式，何謂「Live CD」?
答：參見課本P.14-15
2.何謂命令殼層(shell)?
答：參見課本P.115
3.請說明在終端機內指令: $ cpfile1./subdir1/file2./subdir2之意義為何?
答：參見課本P.128"
`;

exports[`serialize pdf string to paragraphs - 1042_760008_11 1`] = `
"國立空中大學104學年度下學期期中考試題【正參】74
科目：Linux作業系統管理一律橫式作答1頁
一、是非題 25%﹙每題 5分﹚
1.Linux作業系統通常要在大型主機上使用，一般的個人電腦無法安裝使用。(X，1-1)
2.網站提供的Ubuntu下載檔案格式為「ISO」，ISO格式的檔案可以直接燒錄到光碟上。(O，1-4-2)
3.Linux作業系統並不支援圖形化的使用者介面(graphical user interface)，所以這一部分的功能跟微軟公司的視窗作業系統不同。(X，2-1)
4.IE(Internet Explorer)是Ubuntu系統提供的網際網路瀏覽程式。(X，2-4-1)
5.find指令會直接存取硬碟，比較耗費系統資源。(O，3-3.4)
二、選擇題 50%﹙每題 5分﹚
1.Linux中硬碟通常對應到下列哪一種儲存裝置代號？(a) /dev/fd0(b) /dev/sda(c)/dev/hda(c，1-5-2-1)
2.Linux作業系統採用過下列哪一種檔案系統格式？(a) FAT32(b) NTFS(c) ex t4(c，1-5-2-2)
3.Ubuntu利用apt-get及apt-cache兩個指令的變化來進行下列哪一項操作？(a)儲存裝置管理(b)套件管理(c)印表機的管理(b，2-3)
4.假如要改變螢幕的解析度，在Ubuntu的「系統設定」中要選擇哪一項選項進行設定？(a)螢幕(b)外觀(c)主選單(d)印表機(a，2-1-2)
5.Ubuntu中的哪一個指令可以用來移除套件，並且移除與該套件具有相依性之套件？(a) apt-getautoremove(b) apt-get remove(c) apt-get --purge remove(a，2-3-4)
6.使用者登入系統之後，內定的工作目錄稱之為？(a)本目錄(b)家目錄(c)主目錄(b，3-2)
7.下列有關bash的敘述何者錯誤？(a)支援萬用字元(b)具有設定指令別名的功能(c)不可回溯已執行的指令(c，3-2)
8.下列哪一個指令會顯示伺服器上佔用CPU的任務資訊，並且每5秒鐘刷新一次？(a) uptime(b)vmstat(c) top(c，3-5)
9.下列哪一個指令與系統中的程序無關？(a) pstree(b) ps(c) free(c，3-5)
10.下列哪一個指令以分頁方式顯示檔案內容，可以逐頁或逐行往後看？(a) much(b) less(c)more(c，3-3-3)
三、簡答題 25% (每題 5分)
1.何謂「Live CD」？(1-4-1)
2.Linux作業系統與微軟Windows作業系統有什麼相似的地方？(1-4-4)
3.Ubuntu作業系統中使用不同的軟體來安裝軟體，會不會造成管理上的問題？(2-3-7)
4.Ubuntu作業系統中網際網路瀏覽程式需要做哪些設定才能正常使用？(2-4-5)
5.圖形介面組成要素包含哪五部分？(3-1-1)"
`;

exports[`serialize pdf string to paragraphs - 1122_760008_11 1`] = `
"國立空中大學112學年度下學期期中考試題【正參】095
科目：Linux作業系統管理一律橫式作答頁
一、選擇題（每題5分，共50分）
1.下列敘述何者有誤？(A)Ubuntu 作業系統無提供圖形化使用者介面(GUI)的操作方式(B)Linux作業系統可以安裝在個人電腦或大型電腦上(C)Ubuntu官方網站提供下載的檔案格式是「ISO」檔(D) Ubuntu Desktop是可以安裝在個人電腦上的Ubuntu版本。(A，2-1)
2.在 Linux作業系統裡要以文字介面來進行簡單作業，如複製檔案或管理目錄，可以使用下列哪一個程式? (A) Firefox (B)Impress (C)終端機(D)設定值。(C，1-6-3)
3.下列何者是在 VirtualBox虛擬機器軟體上安裝Ubuntu 的好處? (A) 不必再另外準備一台電腦(B)可在一台電腦上安裝多個版本作業系統(C)失敗多次都可以快速重來 (D)以上皆是。(D，1-4)
4.下列何者是作業系統功能裡記憶體的管理範圍? (A)滑鼠(B)執行中的程式與資料(C)中央處理器(D)檔案格式。(B，1-2-1)
5.以下關於自由軟體(Freeware)的敘述，何者錯誤? (A)一般公用授權是免費的(B)允許使用者修改和重新分發軟體的權利(C)使用上無任何限制與規範(D)其中一種授權狀況是不容許程式以商業化的方式使用或散佈。(C，1-1-1)
6.要 Ubuntu系統裡要調整桌面風格或佈景主題時，可在”設定值”裡選擇哪一個選項進行? (A)鍵盤(B)電源(C)外觀(D)隱私。(C，2-1-3)
7.下列哪一個是使用者登入 Ubuntu系統後，內定(預設)的工作目錄？(A)根目前(B)家目錄(C)文件(D)以上皆非。(B，3-2)
8.在 Ubuntu系統裡，下列哪一個功能或指令不是用來管理或安裝套件的? (A)Ubuntu軟體中心(B)apt (C) synaptic (D) cd。(D，2-3)
9.下列何者不是與檔案與目錄管理相關的指令? (A)ps (B)cd (C)cp (D)ls。(A，3-3-2)
10.下列哪 Ubuntu目錄是用來存放Linux 核心及開機程式? (A)/bin (B)/root (C)/boot (D)/etc。(C，1-5)
二、簡答題（每題10分，共50分）
1.請寫出作業系統的四大功能。(1-2-1)
處理器的管理、記憶體的管理、檔案的管理、裝置的管理
2.安裝Ubuntu作業系統有一種方式稱作「Live CD」，請問何謂「Live CD」? (1-4)
3.請簡述Linux作業系統的優點與用途。(1-1-4)
穩定性、輕巧靈活、免費、多人多工的使用環境、硬體規格需求不高、多樣化圖形介面的加入
4.請列出圖形化使用者介面的五項組成要素。(3-1-1)
桌面、視窗、選單、圖示、按鈕
5.請列出至少四個與目錄或檔案相關的Linux指令，並簡單說明其指令功能。(3-3)
pwd、cd、mkdir、rmdir、ls、cp、rm、mv"
`;

exports[`serialize pdf string to paragraphs - 1122_780022_11 1`] = `
"國立空中大學112學年度下學期期中考試題【正參】002
科目：性別、健康與多元文化一律橫式作答1頁
一是非題共10題每題4分共40分**請注意:正確敘述請以O作答錯誤敘述請X作答其他符號作答不計分
O 1.性別運動的目的在於消除性別歧視，讓性別認同、性別特質、外表與裝扮能有多元的流動空間。CH1 P.17
O 2.胸痛是冠狀動脈心臟病男女兩性最常見的症狀。CH5 P.131
X 3.女性主義理論重視性別角色與認知發展的成熟度。CH1 P.8
O 4.本門課的學習目的為能了解性別、健康與多元文化的關係，學習與不同性別者的合作與接納。序
O 5.媽祖繞境透過網路直播讓全世界認識台灣宗教文化，可視為是全球化改變了文化傳遞的方式。CH2 P.43
X 6.CEDAW的全名是消除對男女一切形式歧視公約。CH3 P.55
O 7.實施性別影響評估可以處理性別不平等的議題。CH4 P.87
O 8.一般而言，女性骨折的風險較男性高。CH5 P.137
O 9.性別主流化要求公部門須將性別觀點納入政策規劃與執行及管考。CH3 P.57
X 10.本書討論性別觀點可用性別天生一詞以蔽之。CH1 P.4
二選擇題共10題每題4分共40分
A或C 1.2024巴黎奧運主題之一為性別平等，也是首屆男女選手比例達50%。請問此一進步最直接符合性別主流化哪一項工作？A性別意識培力B性別預算C性別統計CH3 P.61113.4.23更正
A 2.承上題巴黎奧運也符合CEDAW重視的A機會平等B支持平等C優惠平等CH3 P.74
B 3.台灣原住民族正名運動及相關文化復興的運動，是透過何種權利的爭取而讓原住民族得到應有的尊嚴？A政治權B文化權C自治權CH2 P.30
B 4.科技工程領域中性別不平等之處，進行性別分析，利用分析結果納入研發科技以利創新，藉以達到減少性別偏差，改善性別關係，促進性別平等。此稱?A性別統計B性別創新C性別意識培力CH5 P.135
A 5.亞洲非吸菸女性的肺癌環境因子暴露研究指出？A烹飪油煙是重要危險因子B吸菸行為C工廠環境暴露CH5 P.147
B 6.性別影響評估的目的，何者正確？A譴責不平等之處B瞭解不平等的原因C否認性別差異CH4 P.87
B 7.下列關於第三波女性主義的敘述，何者為非？A重視性別多元B反對酷兒運動C重視論述與權力關係CH1 P.14
C 8.從兩性關係到性別關係的敘述何者正確？A重視性別二分B重視性別規範C重視性別多元CH2 P.40
A 9.韓式炸雞近來風靡一時，韓國炸雞店也如雨後春筍般來台成立，但據報載辣度口味亦調整成台灣人可接受程度，請問此為？A全球在地化B經濟全球化C食物全球化CH2 P.44
A10.探討新生兒性別比例差異的社會文化原因，此為？A性別分析B性別統計C性別影響評估CH4P.107
三簡答題共2題每題10分共20分
1.很多人認為醫療是中性的，事實上很多健康議題有其性別差異或是僅以單一性別為主的醫療研究，請舉一例說明性別觀點對於醫療健康議題的重要性。CH5 P.129
答:作答者需舉例說明並呈現性別觀點對於醫療健康的重要性方可得分。
2.為何多元文化主義的核心內涵之一為「消除歧視與不平等對待」?請說明你的看法。CH2 P.30
答:平等對待不同文化及族群團體成為民主國家核心的價值。多元文化的平等概念，除了消除歧視、更強調政治、社會歧視與不平等的消除，包括反對透過同化政策，進行政治社會及經濟權利的壟斷，同時給予弱勢族群較多的資源與權利保障，改善因歧視而導致的不平等。
作答者的回答只要能切合上述內容或言之成理亦可得分。"
`;

exports[`serialize pdf string to paragraphs - 1122_780051_11 1`] = `
"國立空中大學112學年度下學期期中考試題【正參】005
科目：對聯的文學趣味一律橫式作答1頁
壹、簡答題(40%，每題10分)
一、試述構成「對偶」的四項要素。(教科書第5頁；媒體教材1-1-2)
二、中國在哪個時期開始在桃木板上題詞，取代了原本書寫門神名字或掛門神圖像的作法？對聯文化在哪個時代推展到民間？(教科書第15、17頁；媒體教材1-2-3)
三、南明抗清名將黃道周的諷刺聯「史筆留芳，雖未成功終可法；洪恩浩蕩，不能報國反成仇。」用了哪一個修辭手法？另外，「常將有日思無日，莫待無時思有時」又用哪個修辭手法？(教科書第33、35頁；媒體教材1-3-5)
四、清代四川總督丁寶楨十分讚許「十年宦比梅花冷，一夜春隨爆竹來」對聯，試述此聯原本旨意及其比興寓意。(教科書第102-103頁)
貳、申論題(60%，每題20分)
一、「異文比較」是賞析對聯的一種方式。試就「善由此地心無愧，惡過吾門膽自寒」、「善來此地心無愧，惡過我門膽自寒」說明異文含義的不同，並請賞析之。(教科書第94-96頁；媒體教材2-3-9)
二、試逐項檢核並說明「大塊能相假，名山不獨藏。」是否符合基本的對聯格律？另請說明這副對聯適合張貼在什麼場所？為什麼？(教科書第106頁；媒體教材3-2-1)
三、試賞析江蘇無錫陸宣公祠柱聯「世上幾百年舊家，無非積德；天下第一件好事，只是讀書。」(教科書第170-171頁)"
`;
